compiler/noirc_evaluator/src/ssa/opt/simple_optimization.rs

```diff
@@ -95,6 +101,7 @@ pub(crate) struct SimpleOptimizationContext<'dfg, 'mapping> {
     pub(crate) block_id: BasicBlockId,
     pub(crate) instruction_id: InstructionId,
     pub(crate) dfg: &'dfg mut DataFlowGraph,
+    pub(crate) enable_side_effects: ValueId,
```

Comment:
Nice! We could use this in `remove_unreachable_instructions` as well. No need to do it in this PR, I can do it in a follow-up PR.